tests/std/tests/P0355R7_calendars_and_time_zones_dates/test.cpp

```diff
@@ -0,0 +1,930 @@
+    static_assert(noexcept(month{0u}));
+
+    static_assert(noexcept(++m));
+    static_assert(noexcept(m++));
+    static_assert(noexcept(--m));
+    static_assert(noexcept(m--));
+    static_assert(noexcept(m += months{}));
+    static_assert(noexcept(m -= months{}));
+
+    static_assert(noexcept(static_cast<unsigned int>(m)));
+    static_assert(noexcept(m.ok()));
+
+    static_assert(noexcept(m == m));
+    static_assert(noexcept(m <=> m));
+
+    static_assert(noexcept(m + months{}));
+    static_assert(noexcept(months{0} + m));
+
+    static_assert(noexcept(m - months{}));
+    static_assert(noexcept(m - m));
+
+    assert(static_cast<unsigned int>(m) == 1u);
+    assert(m == January);
+    assert(++m == February);
+    assert(m++ == February);
+    assert(m == March);
+
+    assert(--m == February);
+    assert(m-- == February);
+    assert(m == January);
+
+    m += months{2};
+    assert(m == March);
+    m -= months{2};
+    assert(m == January);
+
+    for (unsigned int i = 0; i <= 255; ++i) {
+        if (i >= 1 && i <= 12) {
+            assert(month{i}.ok());
+        } else {
+            assert(!month{i}.ok());
+        }
+    }
+
+    assert(February > m);
+    assert(m < February);
+
+    assert(February + months{11} == January);
+    assert(months{11} + February == January);
+    assert(month{0} + months{1} == January);
+    assert(month{13} + months{1} == February);
+    assert(month{23} + months{1} == December);
+    assert(February - months{2} == December);
+    assert(January - February == months{11});
+}
+
+constexpr void year_test() {
+    year y{1};
+
+    static_assert(noexcept(year{}));
+    static_assert(noexcept(year{0}));
+
+    static_assert(noexcept(++y));
+    static_assert(noexcept(y++));
+    static_assert(noexcept(--y));
+    static_assert(noexcept(y--));
+    static_assert(noexcept(y += years{}));
+    static_assert(noexcept(y -= years{}));
+    static_assert(noexcept(+y));
+    static_assert(noexcept(-y));
+
+    static_assert(noexcept(y.is_leap()));
+    static_assert(noexcept(static_cast<int>(y)));
+    static_assert(noexcept(y.ok()));
+    static_assert(noexcept(y.min()));
+    static_assert(noexcept(y.max()));
+
+    static_assert(noexcept(y == y));
+    static_assert(noexcept(y <=> y));
+
+    static_assert(noexcept(y + years{}));
+    static_assert(noexcept(years{} + y));
+
+    static_assert(noexcept(y - years{}));
+    static_assert(noexcept(y - y));
+
+    static_assert(noexcept(0y));
+
+    assert(static_cast<int>(y) == 1);
+    assert(y == 1y);
+    assert(++y == 2y);
+    assert(y++ == 2y);
+    assert(y == 3y);
+
+    assert(--y == 2y);
+    assert(y-- == 2y);
+    assert(y == 1y);
+
+    y += years{2};
+    assert(y == 3y);
+    y -= years{2};
+    assert(y == 1y);
+
+    year extreme{-30'000};
+    extreme += years{60'000};
+    assert(extreme == 30'000y);
+    extreme -= years{60'000};
+    assert(extreme == -30'000y);
+
+    assert(+y == 1y);
+    assert(-y == -1y);

```

Comment:
```suggestion
    assert(-y == years{-1});
```

`-1y` is `operator-(operator""y(1ULL))`. We probably don't mean to test that.